pkg/steps/artifacts_test.go

```diff
@@ -447,3 +453,126 @@ func TestTestCaseNotifier_SubTests(t *testing.T) {
 		})
 	}
 }
+
+func TestAddArtifactsToTemplateObject(t *testing.T) {
+	corev1Scheme := runtime.NewScheme()
+	utilruntime.Must(coreapi.AddToScheme(corev1Scheme))
+	corev1Codec := serializer.NewCodecFactory(corev1Scheme).LegacyCodec(coreapi.SchemeGroupVersion)
+
+	testCases := []struct {
+		testID         string
+		object         runtime.RawExtension
+		expectedObject runtime.RawExtension
+	}{
+		{
+			testID: "pod object has artifacts volumes/volumeMounts, artifacts container injection expected",
+			object: func() runtime.RawExtension {
+				pod := &coreapi.Pod{
+					TypeMeta:   meta.TypeMeta{Kind: "Pod", APIVersion: "v1"},
+					ObjectMeta: meta.ObjectMeta{Name: "test-pod"},
+					Spec: coreapi.PodSpec{
+						Volumes: []coreapi.Volume{{Name: "artifacts"}},
+						Containers: []coreapi.Container{
+							{
+								Name:         "test",
+								VolumeMounts: []coreapi.VolumeMount{{Name: "artifacts"}},
+							},
+						},
+					},
+				}
+				return runtime.RawExtension{
+					Raw:    []byte(runtime.EncodeOrDie(corev1Codec, pod)),
+					Object: pod.DeepCopyObject(),
+				}
+			}(),
+			expectedObject: func() runtime.RawExtension {
+				pod := &coreapi.Pod{
+					TypeMeta:   meta.TypeMeta{Kind: "Pod", APIVersion: "v1"},
+					ObjectMeta: meta.ObjectMeta{Name: "test-pod"},
+					Spec: coreapi.PodSpec{
+						Volumes: []coreapi.Volume{{Name: "artifacts"}},
+						Containers: []coreapi.Container{
+							{
+								Name:         "test",
+								VolumeMounts: []coreapi.VolumeMount{{Name: "artifacts"}},
+							},
+							makeTestArtifactContainerHelper(),
+						},
+					},
+				}
+				return runtime.RawExtension{
+					Raw:    []byte(runtime.EncodeOrDie(corev1Codec, pod)),
+					Object: pod.DeepCopyObject(),
+				}
+			}(),
+		},
+		{
+			testID: "pod object has no artifacts volumes/volumeMounts, no artifacts container injection expected",
+			object: func() runtime.RawExtension {
+				pod := &coreapi.Pod{
+					TypeMeta:   meta.TypeMeta{Kind: "Pod", APIVersion: "v1"},
+					ObjectMeta: meta.ObjectMeta{Name: "test-pod"},
+					Spec:       coreapi.PodSpec{Containers: []coreapi.Container{{Name: "test"}}},
+				}
+				return runtime.RawExtension{
+					Raw:    []byte(runtime.EncodeOrDie(corev1Codec, pod)),
+					Object: pod.DeepCopyObject(),
+				}
+			}(),
+			expectedObject: func() runtime.RawExtension {
+				pod := &coreapi.Pod{
+					TypeMeta:   meta.TypeMeta{Kind: "Pod", APIVersion: "v1"},
+					ObjectMeta: meta.ObjectMeta{Name: "test-pod"},
+					Spec:       coreapi.PodSpec{Containers: []coreapi.Container{{Name: "test"}}},
+				}
+				return runtime.RawExtension{
+					Raw:    []byte(runtime.EncodeOrDie(corev1Codec, pod)),
+					Object: pod.DeepCopyObject(),
+				}
+			}(),
+		},
+	}
+
+	for _, tc := range testCases {
```

Comment:
more test cases:

 - list of items in the template that contain something other than a Pod along with Pods
 - list of items that contain no Pods
 - Pod that has the volume mount but no containers that use it